opentracing_instrumentation/client_hooks/celery.py

```diff
@@ -0,0 +1,128 @@
+from __future__ import absolute_import
+
+import opentracing
+from opentracing.ext import tags
+
+from ..request_context import get_current_span, span_in_context
+from ._patcher import Patcher
+
+
+try:
+    from celery.app.task import Task
+    from celery.signals import (
+        before_task_publish, task_prerun, task_success, task_failure
+    )
+except ImportError:  # pragma: no cover
+    pass
```

Comment:
not sure I follow  - it will just continue? What happens when it defines install_patches function below, if celery imports failed?

Reply:
> not sure I follow - it will just continue?

Yes, it will.

> What happens when it defines install_patches function below, if celery imports failed?

`CeleryPatcher` become not `applicable` in this case, so `_install_patches` and `_uninstall_patches` [will not be called](https://github.com/Jamim/opentracing-python-instrumentation/blob/feature/celery-upstream/opentracing_instrumentation/client_hooks/_patcher.py#L10-L21).

```bash
(opentracing-python-instrumentation) mim@irma-sun ~/p/i/t/opentracing-python-instrumentation> pip uninstall celery -y
Uninstalling celery-4.3.0:
  Successfully uninstalled celery-4.3.0
(opentracing-python-instrumentation) mim@irma-sun ~/p/i/t/opentracing-python-instrumentation> python
```
```python
Python 3.7.3 (default, Apr  1 2019, 00:04:52) 
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import celery
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'celery'
>>> from opentracing_instrumentation.client_hooks import install_all_patches
>>> install_all_patches()
>>> from opentracing_instrumentation.client_hooks.celery import CeleryPatcher
>>> CeleryPatcher.applicable
False
>>> from opentracing_instrumentation.client_hooks.celery import install_patches
>>> install_patches()
>>>
```

Exactly the same approach is used for `requests` and `boto3`.